libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -421,15 +465,57 @@ bool AP_Arming::battery_checks(bool report)
         }
 
         for (uint8_t i = 0; i < _battery.num_instances(); i++) {
+            // voltage check
             if ((_min_voltage[i] > 0.0f) && (_battery.voltage(i) < _min_voltage[i])) {
                 if (report) {
-                    gcs().send_text(MAV_SEVERITY_CRITICAL, "PreArm: Battery %d voltage %.1f below minimum %.1f",
+                    gcs().send_text(MAV_SEVERITY_CRITICAL, "PreArm: Battery %u voltage %.1f below min %.1f",
                             i+1,
                             (double)_battery.voltage(i),
                             (double)_min_voltage[i]);
                 }
                 return false;
             }
+
+            float temperature = AP_ARMING_BATTERY_DEFAULT_CELSIUS_TEMPERATURE;
+            // temperature minimum check
+            if (static_cast<int16_t>(_min_temperature[i] * 10) > static_cast<int16_t>(AP_ARMING_BATTERY_DEFAULT_CELSIUS_TEMPERATURE * 10)) {
+                if (!_battery.get_temperature(temperature, i)) {
+                    if (report) {
+                        gcs().send_text(MAV_SEVERITY_CRITICAL, "PreArm: No temperature for battery %u",
+                                i + 1);
+                    }
+                    return false;
+                }
+                if (static_cast<int16_t>(temperature * 10) < static_cast<int16_t>(_min_temperature[i] * 10)) {
+                    if (report) {
+                        gcs().send_text(MAV_SEVERITY_CRITICAL, "PreArm: Battery %u temp %.1f below min %.1f",
+                                i + 1,
+                                (double)temperature,
+                                (double)_min_temperature[i]);
+                    }
+                    return false;
+                }
+            }
+
+            // temperature maximum check
+            if (static_cast<int16_t>(_max_temperature[i] * 10) > static_cast<int16_t>(AP_ARMING_BATTERY_DEFAULT_CELSIUS_TEMPERATURE * 10)) {
+                if (static_cast<int16_t>(AP_ARMING_BATTERY_DEFAULT_CELSIUS_TEMPERATURE * 10) == static_cast<int16_t>(temperature * 10) && !_battery.get_temperature(temperature, i)) {
+                    if (report) {
+                        gcs().send_text(MAV_SEVERITY_CRITICAL, "PreArm: No temperature for battery %u",
+                                i + 1);
+                    }
+                    return false;
+                }
+                if (static_cast<int16_t>(temperature * 10) > static_cast<int16_t>(_max_temperature[i] * 10)) {
+                    if (report) {
+                        gcs().send_text(MAV_SEVERITY_CRITICAL, "PreArm: Battery %u temp %.1f below max %.1f",
```

Comment:
This should say the temp is above the max, not below.

Reply:
@WickedShell san. 
I change it to "exceed max".